tools/build_tar/buildtar.go

```diff
@@ -102,6 +102,24 @@ func main() {
 	}
 	defer tf.Close()
 
+	uid := tf.meta.getUID(directory)
```

Comment:
I don't think we need to do this here - `addFile` properly handles creating the parent directories:
https://github.com/kubernetes/repo-infra/blob/069c38ebb042b2a4e3310a9f8c14c5d6972e7401/tools/build_tar/buildtar.go#L227

I think it's `addTar` that doesn't create the parent directories - probably need to extend this case:
https://github.com/kubernetes/repo-infra/blob/069c38ebb042b2a4e3310a9f8c14c5d6972e7401/tools/build_tar/buildtar.go#L275-L279